packages/coinstac-ui/app/render/components/results/displays/result-table.jsx

```diff
@@ -1,28 +1,73 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import {
-  Table,
-} from 'react-bootstrap';
+import { Table, Button } from 'react-bootstrap';
 
+const styles = {
+  csvButton: { position: 'absolute', top: '20rem', right: '2rem' },
+  theading: { color: 'white', fontSize: '0' },
+};
+
+/**
+ * downloadCSV
+ * Create a named empty CSV file inject string blob and download
+ * @param {string} csv Table values
+ * @param {string} filename The name of the file
+ */
+function downloadCSV(csv, filename) {
+  const csvFile = new Blob([csv], { type: 'text/csv' });
+  const downloadLink = document.createElement('a');
+  downloadLink.download = filename;
+  downloadLink.href = window.URL.createObjectURL(csvFile);
+  downloadLink.style.display = 'none';
+  document.body.appendChild(downloadLink);
+  downloadLink.click();
+}
+
+/**
+ * exportTableToCSV
+ * Convert HTML Table to CSV formatted string
+ * @param {string} html Table hypertext
+ * @param {string} filename The name of the file
+ */
+function exportTableToCSV(html, filename) {
+  const csv = [];
+  const rows = document.querySelectorAll('table tr');
```

Comment:
Do you need DOM APIs to retrieve the data? If your component already has all access to the renderable data it should be able to map it into a CSV. Just a thought.

Reply:
This is true, however the goal was to preserve the unique table formatting for the end user, which would have to be fully duplicated if I pulled from the results object.